### PR TITLE
WM4-Kavita-Patil-JavaScript-Core-1-Coursework-Week4

### PR DESCRIPTION
**Pull request by kavitappatil**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

-----
[View rendered exercises/G-numbers/README.md](https://github.com/kavitappatil/JavaScript-Core-1-Coursework-Week1/blob/master/exercises/G-numbers/README.md)
[View rendered exercises/L-functions-nested/README.md](https://github.com/kavitappatil/JavaScript-Core-1-Coursework-Week1/blob/master/exercises/L-functions-nested/README.md)

### CODE REVIEW

File: mandatory/1-syntax-errors.js [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 
-function introduceMe(name, age)
-  return "Hello, my name is " + name "and I am " age + "years old";
+function introduceMe(name, age) {
+  return "Hello, my name is " + name + " and I am " + age + " years old";
+}
 
 function getTotal(a, b) {
-  total = a ++ b;
+  total = a + b;
```
Comment (jdbevan) [inline mandatory/1-syntax-errors.js:12]: Do you remember what we need to put at the start of the line when defining or creating a new variable?

File: mandatory/1-syntax-errors.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 
-  return "The total is total";
+  return `The total is ${total}`;
```
Comment (jdbevan) [inline mandatory/1-syntax-errors.js:14]: Nice use of string interpolation!

File: mandatory/3-function-output.js [hunk 3]
```diff
@@ -1,16 +1,20 @@
 // Add comments to explain what this function does. You're meant to use Google!
 function getRandomNumber() {
+
+  //returns a random number between 0 (included) and 1 (excluded) and multiple by 10:
```
Comment (jdbevan) [inline mandatory/3-function-output.js:4]: Do you know what kind of random number? Integer? Decimal? Negative or positive?

File: mandatory/4-tax.js [hunk 4]
```diff
@@ -5,7 +5,10 @@
   Sales tax is 20% of the price of the product.
 */
 
-function calculateSalesTax() {}
+function calculateSalesTax(price) {
+  var salesTax = price + price * .20;
```
Comment (jdbevan) [inline mandatory/4-tax.js:9]: This confused me to begin with as to why you were multiplying the price by 0.2. One of the reasons is that I have no idea what the significance of that number is. In the industry we'd refer to this as a "magic number". It's considered better to treat it as a named constant defined somewhere in your code. Some people/organisations will capitalize shared constants, something like:

```
const TAX_RATE = 0.2;

function calculateSalesTax(price) {
    let tax = price * TAX_RATE;
    return price + tax;
}
```

File: mandatory/1-syntax-errors.js (hunk 1 above)
Comment (kavitappatil) [reply mandatory/1-syntax-errors.js:12]: I need to add var or let

File: mandatory/3-function-output.js (hunk 3 above)
Comment (kavitappatil) [reply mandatory/3-function-output.js:4]: It always returns a number lower than 1.

File: mandatory/4-tax.js (hunk 4 above)
Comment (kavitappatil) [reply mandatory/4-tax.js:9]: oh, ok thanks

File: mandatory/3-function-output.js (hunk 3 above)
Comment (jdbevan) [reply mandatory/3-function-output.js:4]: Could it return -100? Or 0.56?

**Pull request closed without merging**